### PR TITLE
Enable integration tests in build-script-helper.py

### PR DESCRIPTION
**Pull request by porglezomp**
This will enable the swift build-script test runs to get a more complete run—"enable integration tests" seems in-line with the scope of other projects test runs.

### CODE REVIEW

**Comment by cltnschlosser** [comment]
Pretty sure the only integration tests we have are lit tests which also requires `SWIFT_DRIVER_LIT_DIR` to be set. These aren't passing yet, so don't enable them for swift PRs if it's going to break the build.

**Comment by porglezomp** [comment]
Hm, are there no other integration tests at all? I was planning on adding the `SWIFT_DRIVER_LIT_DIR` separately while I start getting those passing. This is part of the lead-up work to getting those tests enabled for PR testing.

**Comment by porglezomp** [comment]
@swift-ci please test

**Comment by porglezomp** [comment]
Added a `--lit-test-dir` flag that will forward to the environment variable. Since nothing passes this flag yet, it won't impact CI runs.

**Comment by porglezomp** [comment]
@swift-ci please test

**Review by artemcm — APPROVED** [review]

File: Utilities/build-script-helper.py [hunk 1]
```diff
@@ -191,6 +191,9 @@ def handle_invocation(args):
     test_args += ['-Xswiftc', '-enable-testing']
     if should_test_parallel():
       test_args += ['--parallel']
+    env['SWIFT_DRIVER_ENABLE_INTEGRATION_TESTS'] = "1"
```
Comment (shahmishal) [inline Utilities/build-script-helper.py:194]: Can we add a comment on why this is getting added here?

Comment (porglezomp) [reply Utilities/build-script-helper.py:195]: Added a comment.

**Review by shahmishal — APPROVED** [review]

**Comment by porglezomp** [comment]
@swift-ci please smoke test and merge

**Comment by porglezomp** [comment]
@swift-ci please test and merge

**Comment by edymtt** [comment]
Would it be possible to also run Swift PR testing (smoketest and toolchain)?
This can be done by using an empty PR against the Swift repo, e.g. https://github.com/apple/swift/pull/34782
See also https://github.com/apple/swift/blob/main/docs/ContinuousIntegration.md#cross-repository-testing

File: Utilities/build-script-helper.py [hunk 2]
```diff
@@ -191,6 +191,10 @@ def handle_invocation(args):
     test_args += ['-Xswiftc', '-enable-testing']
     if should_test_parallel():
       test_args += ['--parallel']
+    # The test suite consults these variables to control what tests get run
```
Comment (edymtt) [inline Utilities/build-script-helper.py:194]: (optional) The comment could expand on why in this situation it's ok to opt in into those tests (basically the first part of the PR description)

**Review by edymtt — APPROVED** [review]

**Comment by porglezomp** [comment]
@swift-ci please test

**Pull request merged**